bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/capability/RefreshCapability.java

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.netatmo.internal.handler.capability;
+
+import static java.time.temporal.ChronoUnit.SECONDS;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Optional;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.netatmo.internal.api.dto.NAThing;
+import org.openhab.binding.netatmo.internal.handler.CommonInterface;
+import org.openhab.core.thing.ThingStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@link RefreshCapability} is the class used to embed the refreshing needs calculation for devices
+ *
+ * @author Gaël L'hopital - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class RefreshCapability extends Capability {
+    private static final Duration DEFAULT_DELAY = Duration.of(20, SECONDS);
+    private static final Duration PROBING_INTERVAL = Duration.of(120, SECONDS);
+
+    private final Logger logger = LoggerFactory.getLogger(RefreshCapability.class);
+    private final ScheduledExecutorService scheduler;
+
+    private Duration dataValidity;
+    private Instant dataTimeStamp = Instant.now();
+    private @Nullable Instant dataTimeStamp0;
```

Comment:
Perhaps `Instant.MIN` could be used to avoid `@Nullable`. @lolodomo - consider for your pending PR.

Reply:
Done.